GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using UnityEngine;
+
+namespace Turn_based_assets.Scripts
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
+        [SerializeField] private GameObject cube;
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        private Plane _plane;
+        private Vector3 distanceFromCamera;
+        
+
+        private void Start()
+        {
+            var position = Camera.main.transform.position;
```

Comment:
`Camera.main` is expensive. Consider cheaper alternatives to getting the camera.